### DISCUSSION TITLE
How do i make a discord bot reply multiple responses to multiple commands

### DISCUSSION THREAD

**Discussion by tommyboi1** [discordjs/discord.js#6069] [Q&A] [answered]
Problem: I am trying to make a discord bot in VSCode with discord.js and I want it to respond to me if I say something but for it to not always be the same as the last time, here is an example,

me: Hello or Hi or Sup or Yo (case insensitive)

bot: Sup or Hello or Hi or Yo (case insensitive)

this is my code so far:

> let repliesHi = [“Hi”, “Hello”, “Yo”, “Sup”];
> let randomHi = Math.floor(Math.random() * repliesHi.length);

> client.on("message", message => {
>     if (/\bhi\b/i.test(message.content)) {
>         if (message.author.bot) return;
>         message.channel.send(repliesHi[randomHi]);
>     }
> });

any help would be greatly appreciated

edit: without having to restart the program every time I want a different answer

**Comment by lavgup** [comment]
Your regex test doesn't cater for all possible inputs. You could just check if the message content is an element of the array you have initialised rather than using regex (and use `String` methods to support case insensitivity).

**Comment by monbrey** [comment]
You are generating your random number once, and storing it as a constant. This never changes, so it will always refer to the same response. Generate a new random number each message event.

**Comment by Cats3153** [comment, marked as answer]
call the random function inside `client.on`

```js
let repliesHi = ["Hi", "Hello", "Yo", "Sup"];

client.on("message", message => {
	if (message.author.bot) return;
	if (/\bhi\b/i.test(message.content)) {
		let randomHi = Math.floor(Math.random() * repliesHi.length);
		message.channel.send(repliesHi[randomHi]);
	}
});
```

this way, each time the callback function for the `message` event is called, a new random number would be generated, instead of having it be generated one time at the beginning of the file
